rs/registry/canister/src/mutations/subnet.rs

```diff
@@ -253,7 +256,9 @@ impl Registry {
 
         ChainKeyInitialization {
             key_id: Some((&dealing_request.key_id).into()),
-            dealings: Some(response.initial_dkg_dealings),
+            initialization: Some(chain_key_initialization::Initialization::Dealings(
```

Comment:
Does the mainnet replica version on the NNS subnet already support the new `initialization` field?

Reply:
The wire representation of the `dealings` variant does not change, as long as the NNS does not start to create `transcript` variants (which it does not), there is no need to update the replica first.

This PR contains both updates, for the NNS and for the replica. It does not matter in which order they are being released.

Or what exactly do you mean by support?

Reply:
> as long as the NNS does not start to create transcript variants (which it does not),

This was the part that's not obvious to me, without analyzing the whole code. 

I trust that you are taking care to comply with this requirement. Still, it could be good to mention in the PR description, that this PR doesn't make the Registry create records with `ChainKeyInitialization.initialization`, and that that's presumably planned in a follow-up, right?

Reply:
I will mention it in the description.

Just for clarification:
The old  `ChainKeyInitialization.dealings` and the new `ChainKeyInitializaiton.initializaition.dealings` look exactly the same when serialized. Only the client code changes, since `dealings` is no longer a field but a variant of the `oneof` `initialization`. This means that this PR compatible in both directions.

Only once we start creating `ChainKeyInitializaiton.initializaition.transcript` variants in the registry will we need the replica to understand the new variant. It will be quite some time until we will start writing that code, since a lot of different pieces have to be place, including the consensus code to handle these transcripts.

This PR is exactly to unblock us from starting our work on the consensus code.

Reply:
Thanks for the clarification. 

Which registry operations would fail if there somehow were to attempt creating `ChainKeyInitializaiton.initializaition.transcript` records too soon? Is there a Jira ticket that I can search for in the code to find those places?

Reply:
There is a new variant of `MasterPublicKeyId` (`::VetKD`), which was added in #2108 .

Right now, if we would try to Create/Update/Recover a Subnet with a new VetKey, we would pass the `MasterPublicKey::VetKD` key into `compute_initial_i_dkg_dealings`, which would then fail which in turn would fail the call to the registry.

In the future, we would recognise this case, call a (yet to be implemented) new management endpoint that would then give us the transcript that we then put into the new `transcript` variant of the CUP.
So right now there is no way for the registry to make a valid `transcript` anyway.